Oracle.md

```diff
@@ -0,0 +1,135 @@
+# Oracle specifications
+
+## Introduction
+
+For the purpose of these specifications, an event is a digital representation of a real work fact.
+An [oracle](./Introduction#Oracle) is an entity that commits to publishing a signature over a (number of) event outcome(s) ahead of time by releasing an [R-value](./Introduction#R-value) as well as necessary information for two parties to build a set of [CETs](./Introduction#Contract-Execution-Transaction-(CET)).
+This information is committed in a so-called _event template_ that will be further detailed in this document.
+Templates are classified and discovered through a namespace presently represented as Unique Resource Identifier (URI) paths.
+
+## Event templates
+
+An event template provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Regardless of the outcome type, an event template should include:
+
+- The R value that will be used to sign the event outcome (possibly several).
+- The public key of the oracle.
```

Comment:
I expect that the public key is cached or even embedded in the app and not communicated with the R value (the public key is static and should only be fetched once).

I think "event announcement" would be a better name for this with the template being the name for the part of the announcement that describes the outcomes (this caused me confusion below). I'm not sure if "template" is right. Usually a template is something you fill in. Maybe just "outcome description" even though that's not great either.

Reply:
I see the point about the public key, at first I was thinking similarly, but it is also really convenient that everything is self contained. Otherwise we also need to standardize a way to retrieve public keys (not that it's impossible though).

And I agree that template is probably not a good name, I think originally we used "template" to describe something that an oracle would fill in to describe an event, but given how I presented things it definitely doesn't fit. Also are you saying that it should be divided in two parts (announcement and description) or am I misunderstanding?

Reply:
Went with "event descriptor" for now, still not super happy with it though.